lib/storage/src/chunks/getChunkStream.ts

```diff
@@ -0,0 +1,43 @@
+import { RawDataPart } from "../Upload";
+import { Buffer } from "buffer";
+
+interface Buffers {
+  chunks: Buffer[];
+  length: number;
+}
+
+export async function* getChunk<T>(
+  data: T,
+  partSize: number,
+  getNextData: (data: T) => AsyncGenerator<Buffer>
+): AsyncGenerator<RawDataPart> {
```

Comment:
Updates:
* value `void` for `TReturn`: The generator function falls off the end without a return and the caller gets back `{ value: undefined, done: true }`.
* value `undefined` for `TNext`: next function does not accept any parameters.
```suggestion
): AsyncGenerator<RawDataPart, void, undefined> {
```